### ISSUE TITLE
Recursive call of 'Thread_base::_alloc_context()'

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode#1244]
When running the ```lwip_lx``` test of the current staging branch on 64-Bit Linux, the timer service gets stuck in a recursive call of ```Thread_base::_alloc_context()```, caused by RAM quota depletion during the main thread context allocation, which leads to the creation of a signal handler thread, which involves another call of ```Thread_base::_alloc_context()```.

Backtrace:

```
#0  pseudo_end () at /home/christian/projects/genode/src/genode/repos/base-linux/../base-linux/src/platform/x86_64/lx_syscall.S:29
#1  0x000000000100d04f in lx_futex (uaddr=0x105db60, op=0, val=0) at /home/christian/projects/genode/src/genode/repos/base-linux/src/platform/linux_syscalls.h:319
#2  0x000000000100d131 in thread_stop_myself () at /home/christian/projects/genode/src/genode/repos/base-linux/src/base/lock/lock_helper.h:74
#3  0x000000000100d315 in Genode::Cancelable_lock::lock (this=0x10869e0) at /home/christian/projects/genode/src/genode/repos/base/src/base/lock/lock.cc:112
#4  0x0000000001000018 in Genode::Lock::lock (this=0x10869e0) at /home/christian/projects/genode/src/genode/repos/base/include/base/lock.h:35
#5  0x0000000001000862 in Genode::Lock_guard<Genode::Lock>::Lock_guard (this=0x10921e0, lock=...) at /home/christian/projects/genode/src/genode/repos/base/include/base/lock_guard.h:40
#6  0x0000000001022a93 in Genode::Thread_base::_alloc_context (this=0x105d9a0, stack_size=32768, main_thread=false) at /home/christian/projects/genode/src/genode/repos/base/src/base/thread/thread.cc:78
#7  0x0000000001022ef4 in Genode::Thread_base::Thread_base (this=0x105d9a0, name=0x1039945 "signal handler", stack_size=32768, type=Genode::Thread_base::NORMAL, cpu_session=0x0) at /home/christian/projects/genode/src/genode/repos/base/src/base/thread/thread.cc:200
#8  0x0000000001022f98 in Genode::Thread_base::Thread_base (this=0x105d9a0, name=0x1039945 "signal handler", stack_size=32768, type=Genode::Thread_base::NORMAL) at /home/christian/projects/genode/src/genode/repos/base/src/base/thread/thread.cc:208
#9  0x0000000001012c47 in Genode::Thread<32768u>::Thread (this=0x105d9a0, name=0x1039945 "signal handler", type=Genode::Thread_base::NORMAL) at /home/christian/projects/genode/src/genode/repos/base/include/base/thread.h:520
#10 0x0000000001011969 in Signal_handler_thread::Signal_handler_thread (this=0x105d9a0) at /home/christian/projects/genode/src/genode/repos/base/src/base/signal/signal.cc:64
#11 0x0000000001011ad1 in signal_handler_thread () at /home/christian/projects/genode/src/genode/repos/base/src/base/signal/signal.cc:91
#12 0x0000000001011faa in Genode::Signal_receiver::Signal_receiver (this=0x105dee0) at /home/christian/projects/genode/src/genode/repos/base/src/base/signal/signal.cc:264
#13 0x0000000001018ba0 in resource_sig_rec () at /home/christian/projects/genode/src/genode/repos/base/src/base/env/env.cc:36
#14 0x0000000001018cdb in Genode::Expanding_parent_client::_fallback_sig_cap (this=0x1086960) at /home/christian/projects/genode/src/genode/repos/base/src/base/env/env.cc:67
#15 0x000000000101de36 in Genode::Expanding_parent_client::resource_request (this=0x1086960, args=...) at /home/christian/projects/genode/src/genode/repos/base/src/base/env/platform_env_common.h:302
#16 0x0000000001018485 in Genode::Expanding_ram_session_client::alloc(unsigned long, Genode::Cache_attribute)::{lambda()#2}::operator()() const (__closure=0x1092590) at /home/christian/projects/genode/src/genode/repos/base/src/base/env/platform_env_common.h:123
#17 0x0000000001019bc3 in retry<Genode::Ram_session::Quota_exceeded, Genode::Expanding_ram_session_client::alloc(unsigned long, Genode::Cache_attribute)::{lambda()#1}, Genode::Expanding_ram_session_client::alloc(unsigned long, Genode::Cache_attribute)::{lambda()#2}>(Genode::Expanding_ram_session_client::alloc(unsigned long, Genode::Cache_attribute)::{lambda()#1}, Genode::Expanding_ram_session_client::alloc(unsigned long, Genode::Cache_attribute)::{lambda()#2}, unsigned int) (func=..., handler=..., attempts=2) at /home/christian/projects/genode/src/genode/repos/base/src/base/env/platform_env_common.h:54
#18 0x0000000001018502 in Genode::Expanding_ram_session_client::alloc (this=0x105df78, size=131072, cached=Genode::CACHED) at /home/christian/projects/genode/src/genode/repos/base/src/base/env/platform_env_common.h:125
#19 0x0000000001022b81 in Genode::Thread_base::_alloc_context (this=0x1092840, stack_size=131072, main_thread=true) at /home/christian/projects/genode/src/genode/repos/base/src/base/thread/thread.cc:109
#20 0x0000000001022ef4 in Genode::Thread_base::Thread_base (this=0x1092840, name=0x103c480 "main", stack_size=131072, type=Genode::Thread_base::MAIN, cpu_session=0x0) at /home/christian/projects/genode/src/genode/repos/base/src/base/thread/thread.cc:200
#21 0x0000000001022f98 in Genode::Thread_base::Thread_base (this=0x1092840, name=0x103c480 "main", stack_size=131072, type=Genode::Thread_base::MAIN) at /home/christian/projects/genode/src/genode/repos/base/src/base/thread/thread.cc:208
#22 0x0000000001032c95 in Genode::Thread<131072u>::Thread (this=0x1092840, name=0x103c480 "main", type=Genode::Thread_base::MAIN) at /home/christian/projects/genode/src/genode/repos/base/include/base/thread.h:520
#23 0x0000000001032b3c in Main_thread::Main_thread (this=0x1092840, reinit=false) at /home/christian/projects/genode/src/genode/repos/base/src/platform/init_main_thread.cc:54
#24 0x0000000001032b9b in main_thread () at /home/christian/projects/genode/src/genode/repos/base/src/platform/init_main_thread.cc:67
#25 0x0000000001032c46 in init_main_thread () at /home/christian/projects/genode/src/genode/repos/base/src/platform/init_main_thread.cc:122
#26 0x0000000001032a81 in _start () at /home/christian/projects/genode/src/genode/repos/base/src/platform/x86_64/crt0.s:48
```

**Comment by chelmuth** [comment]
Is this issue still valid?

**Comment by cproc** [comment]
The issue was fixed by f11b0439b22402af59a3f2e63c33c9954378b911.
